### ISSUE TITLE
dashboard record view fails due to numeric values

### ISSUE THREAD

**Issue by RayPlante** [MDCS-community/MDII-issues#6]
## Summary

Under certain circumstances, attempting to view a resource shown in the dashboard (under "My Resources") by clicking on its "View" link (`http://localhost/dashboard/detail?type=record&id=...`) will instead result in an internal error with the message, 

   `coercing to Unicode: need string or buffer, int found`

The error occurs apparently because the XML data is being incorrectly saved into its MongoDB document.  I have determined what the failure is; however, I have not figured out how it happens.

## How to replicate

This bug is known to occur under the `bipm` branch; however, I expect that given similar schemas, it can manifest itself on all branches (including `master`).  Furthermore, it is most reliably replicated when the resource record is created by uploading the record as an XML file; however, I believe I have seen this bug occur under other circumstances.  One should use the `bipm` branch to replicate this error so as to use a schema with the features needed to produce the error.  Unpack attached file, [sample-org-resource-rec.zip](https://github.com/MDCS-community/MDII-issues/files/407641/sample-org-resource-rec.zip), to find a sample resource document, `nist.xml`.  Load this file to as an "Institution" (Organization) to see the error.  Be sure to save the data and add it as a resource.

Once the sample record is saved as a resource by attempting to view the record by clicking on the "View" link that is part of its listing in the list of resources, accessed via the dashboard.  This will produce the error page.

## Analysis

The error page returned with attempting to the view the loaded record gives a stack trace for the error.  The error occurs when attempting to convert the record retrieved from the MongoDB `xmldata` collection back into XML; this happens within `user_dashboard/views.py` (line 224):

     xmlString = xmltodict.unparse(xmlString['content']).encode('utf-8')

The `xmltodict` module produces the actual error message.  The specific piece of data that is causing the error looks like this:

    OrderedDict(u'@xmlns': u'', u'#text': 1901)

The problem is that `xmltodict` is expecting `1901` to be string when it is realized as an integer.  

A quick way to reproduce this xmltodict error (independent of MDCS) is like this:

    import xmltodict
    data = { 'date': { '@type': 'Created', '#text': 1901 } }
    xml = xmltodict.unparse(data)

This results in the following error:

     Traceback (most recent call last):
       File "<stdin>", line 1, in <module>
       File "/usr/local/lib/python2.7/dist-packages/xmltodict.py", line 325, in unparse
         _emit(key, value, content_handler, **kwargs)
       File "/usr/local/lib/python2.7/dist-packages/xmltodict.py", line 295, in _emit
         content_handler.characters(cdata)
       File "/usr/lib/python2.7/xml/sax/saxutils.py", line 187, in characters
         content = unicode(content, self._encoding)
     TypeError: coercing to Unicode: need string or buffer, int found

What is not clear is how and where the original data imported as XML is converted into its dictionary/JSON format for saving into the database.  How ever this is done, the date value, "1901", is incorrectly be saved as an integer rather than a string.  Changing this value to a string directly in the MongoDB database corrects the problem, allowing the resource to be viewed without error.  

It is worth noting that the underlying schema does _not_ define the value of the date field to be an integer; rather, it is defined to be an `xsd:string` with a restricted pattern.  (See `static/resources/xsd/Organization.xsd` in the `bipm` branch).

**Comment by RayPlante** [comment]
I discovered the cause of the bug.  `mgi/models.py` was updated with parallel pre/postprocessor functions for `xmltodict`; however, line 224 in `user_dashboard/views.py` was not updated to use the needed preprocessor function.

This bug can be fixed by changing line 224 to the following:

    xmlString = XMLdata.unparse(xmlString['content']).encode('utf-8')

as `XMLdata.unparse()` uses the preprocessor transparently.
